### DISCUSSION TITLE
Using more that 2 CE pins with SPI, is it possible with this library ?

### DISCUSSION THREAD

**Discussion by stephend61** [Pi4J/pi4j#399] [Q&A] [answered]
Hi, I am trying to connect 5 MCP3008 chips to a raspbery pi 4 B, but can't work out how to address them all, has anyone else done something similar  ?

I have added the extra chipselect entries, but I can't see  a way of using them with this library, is it possible ?
Or maybe some other way, preferably with Java.

I have it working with 2 at the meoment.

ls -la /dev/spi*
crw-rw---- 1 root spi 153, 3 Oct 20 22:17 /dev/spidev0.0
crw-rw---- 1 root spi 153, 4 Oct 20 22:17 /dev/spidev0.1
crw-rw---- 1 root spi 153, 2 Oct 20 22:17 /dev/spidev0.2
crw-rw---- 1 root spi 153, 1 Oct 20 22:17 /dev/spidev0.3
crw-rw---- 1 root spi 153, 0 Oct 20 22:17 /dev/spidev0.4

Thanks.
Regards
Stephen.

**Comment by stephend61** [comment, marked as answer]
Don't worry I worked it out, I now have all 5 working using pi4j, thanks.

**Reply by FDelporte** [reply]
Hi @stephend61 , how did you fix, is it something we should document on pi4j.com?

**Comment by stephend61** [comment]
I ended up just using gpio pins to control the chip select and ignored the 2 default select pins, since I need 5 select pins and couldn't work out how to expand what pi4j had available, if there is a better way please let me know.

Thanks.

**Reply by taartspi** [reply]
At present time that is the simple way to have more than 2 CE lines.  If gpios are limited a gpio expander chip via i2c is usable although i wonder if its operation would be slower.  At any rate the present pigpio spi provider has these limits.  We are working on a new spi provider as it is required for the model Pi5.   We will make sure we do not have this limitation.

**Reply by stephend61** [reply]
Sounds good might eventually move to the Pi 5, so will look at making the change then, thanks
